packages/@aws-cdk/aws-ecs-patterns/lib/base/application-multiple-target-groups-service-base.ts

```diff
@@ -0,0 +1,612 @@
+   */
+  public readonly desiredCount: number;
+
+  /**
+   * The default Application Load Balancer for the service (first added load balancer).
+   */
+  public readonly loadBalancer: ApplicationLoadBalancer;
+
+  /**
+   * The default listener for the service (first added listener).
+   */
+  public readonly listener: ApplicationListener;
+
+  /**
+   * The cluster that hosts the service.
+   */
+  public readonly cluster: ICluster;
+
+  protected logDriver?: LogDriver;
+  protected listeners = new Array<ApplicationListener>();
+  protected targetGroups = new Array<ApplicationTargetGroup>();
+
+  private loadBalancers = new Array<ApplicationLoadBalancer>();
+
+  /**
+   * Constructs a new instance of the ApplicationMultipleTargetGroupsServiceBase class.
+   */
+  constructor(scope: cdk.Construct, id: string, props: ApplicationMultipleTargetGroupsServiceBaseProps = {}) {
+    super(scope, id);
+
+    this.validateInput(props);
+
+    this.cluster = props.cluster || this.getDefaultCluster(this, props.vpc);
+    this.desiredCount = props.desiredCount || 1;
+    if (props.taskImageOptions) {
+      this.logDriver = this.createLogDriver(props.taskImageOptions.enableLogging, props.taskImageOptions.logDriver);
+    }
+
+    if (props.loadBalancers) {
+      for (const lbProps of props.loadBalancers) {
+        const lb = this.createLoadBalancer(lbProps.name, lbProps.publicLoadBalancer);
+        this.loadBalancers.push(lb);
+        const protocolType = new Set<ApplicationProtocol>();
+        if (lbProps.listeners) {
+          for (const listenerProps of lbProps.listeners) {
+            const protocol = this.createListenerProtocol(listenerProps.protocol, listenerProps.certificate);
+            if (listenerProps.certificate !== undefined && protocol !== undefined && protocol !== ApplicationProtocol.HTTPS) {
+              throw new Error('The HTTPS protocol must be used when a certificate is given');
+            }
+            protocolType.add(protocol);
+            const listener = this.configListener(protocol, {
+              certificate: listenerProps.certificate,
+              domainName: lbProps.domainName,
+              domainZone: lbProps.domainZone,
+              listenerName: listenerProps.name,
+              loadBalancer: lb,
+              port: listenerProps.port
+            });
+            this.listeners.push(listener);
+          }
+        }
+        const domainName = this.createDomainName(lb, lbProps.domainName, lbProps.domainZone);
+        new cdk.CfnOutput(this, `LoadBalancerDNS${lb.node.id}`, { value: lb.loadBalancerDnsName });
+        for (const protocol of protocolType) {
+          new cdk.CfnOutput(this, `ServiceURL${lb.node.id}${protocol.toLowerCase()}`, { value: protocol.toLowerCase() + '://' + domainName });
+        }
+      }
+      if (this.loadBalancers.length === 0) {
+        throw new Error('At least one load balancer should be specified');
+      }
+      this.loadBalancer = this.loadBalancers[0];
```

Comment:
Nit: It might be helpful to add a comment both here and in line 404 that this is where you are setting the default LB and listener?